app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,41 @@
+package org.mozilla.fenix.ext
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.BookmarkNodeType
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class BookmarkNodeTest {
+
+    @Test
+    fun `minus`() {
+        val nodeType = BookmarkNodeType.ITEM
+        // Create children nodes
+        val bookMarkChild1 = BookmarkNode(nodeType, "12", "01", 1, "Testing 1", "testing1.url", null)
+        val bookMarkChild2 = BookmarkNode(nodeType, "13", "02", 2, "Testing 2", "testing2.url", null)
+
+        // Create list of children belonging to bookMarkNode
+        val childrenNodes = listOf(bookMarkChild1, bookMarkChild2)
+
+        // Create a set of children nodes to pass to minus function
+        val childrenSubsetNodes = setOf(bookMarkChild1)
+
+        // Create a list of the remainders
+        val minusSubsetNodes = listOf(bookMarkChild2)
+
+        // Create the node to test
+        var bookmarkNode = BookmarkNode(nodeType, "11", "00", 0, "Testing", "testing.url", childrenNodes)
```

Comment:
nit: there's no need to use var here. You could just create a new variable instead of overriding and you're (usually) less likely to run into problems that way. Avoid mutability unless it's really the easiest way to express something (and think thrice about that :D).